add doc for the Huawei OBS new bindings component (#2447)
dapr/docs: daprdocs/content/en/reference/components-reference/supported-bindings/huawei-obs.md

---
type: docs
title: "Huawei OBS binding spec"
linkTitle: "Huawei OBS"
description: "Detailed documentation on the Huawei OBS binding component"
aliases:
  - "/operations/components/setup-bindings/supported-bindings/huawei-obs/"
---

## Component format

> To setup Huawei OBS (output) binding create a component of type `bindings.huawei.obs`. See [this guide]({{< ref "howto-bindings.md#1-create-a-binding" >}}) on how to create and apply a binding configuration.

Reviewer: ```suggestion
To setup Huawei Object Storage Service (OBS) (output) binding create a component of type `bindings.huawei.obs`. See [this guide]({{< ref "howto-bindings.md#1-create-a-binding" >}}) on how to create and apply a binding configuration.
```

  ```bash
  curl -d "{ \"operation\": \"create\", \"data\": \"Hello World\" }" http://localhost:<dapr-port>/v1.0/bindings/<binding-name>
  ```
  {{% /codetab %}}

  {{% codetab %}}
  ```bash
  curl -d '{ "operation": "create", "data": "Hello World" }' \
        http://localhost:<dapr-port>/v1.0/bindings/<binding-name>
  ```
  {{% /codetab %}}

{{< /tabs >}}

##### Save text to a specific file

{{< tabs Windows Linux >}}

  {{% codetab %}}
  ```bash
  curl -d "{ \"operation\": \"create\", \"data\": \"Hello World\", \"metadata\": { \"key\": \"my-test-file.txt\" } }" \
        http://localhost:<dapr-port>/v1.0/bindings/<binding-name>
  ```
  {{% /codetab %}}

  {{% codetab %}}
  ```bash
  curl -d '{ "operation": "create", "data": "Hello World", "metadata": { "key": "my-test-file.txt" } }' \
        http://localhost:<dapr-port>/v1.0/bindings/<binding-name>
  ```
  {{% /codetab %}}

{{< /tabs >}}

#### Response

The response JSON body contains the `statusCode` and the `versionId` fields. The `versionId` will have a value returned only if the bucket versioning is enabled and an empty string otherwise.

### Upload file

> To upload a binary file (e.g., _.jpg_, _.zip_), invoke the Huawei OBS binding with a `POST` method and the following JSON body:

Reviewer: ```suggestion
To upload a binary file (for example, _.jpg_, _.zip_), invoke the Huawei OBS binding with a `POST` method and the following JSON body:
```

> > Note: by default, a random UUID is generated, if we don't specify the `key`. See the example below for Metadata support to set the destination file name. This API can still be used to upload a regular file (i.e., plain text).

Reviewer: ```suggestion
> Note: by default, a random UUID is generated, if you don't specify the `key`. See the example below for metadata support to set the destination file name. This API can be used to upload a regular file, such as a plain text file).
```

  ```
  {{% /codetab %}}

  {{% codetab %}}
  ```bash
  curl -d '{ "operation": "get", "metadata": { "key": "my-test-file.txt" }}' \
        http://localhost:<dapr-port>/v1.0/bindings/<binding-name>
  ```
  {{% /codetab %}}

{{< /tabs >}}

#### Response

The response body contains the value stored in the object.

### Delete object

To perform a delete object operation, invoke the Huawei OBS binding with a `POST` method and the following JSON body:

```json
{
  "operation": "delete",
  "metadata": {
    "key": "my-test-file.txt"
  }
}
```

The metadata parameters are:

- `key` - the name of the object

#### Examples

##### Delete object

{{< tabs Windows Linux >}}

  {{% codetab %}}
  ```bash
  curl -d '{ \"operation\": \"delete\", \"metadata\": { \"key\": \"my-test-file.txt\" }}' http://localhost:<dapr-port>/v1.0/bindings/<binding-name>
  ```
  {{% /codetab %}}

  {{% codetab %}}
  ```bash
  curl -d '{ "operation": "delete", "metadata": { "key": "my-test-file.txt" }}' \
        http://localhost:<dapr-port>/v1.0/bindings/<binding-name>
  ```
  {{% /codetab %}}

{{< /tabs >}}

#### Response

> An HTTP 204 (No Content) and empty body will be retuned if successful.

Reviewer: ```suggestion
An HTTP 204 (No Content) and empty body are returned if successful.
```

### List objects

To perform a list object operation, invoke the Huawei OBS binding with a `POST` method and the following JSON body:

```json
{
  "operation": "list",
  "data": {
    "maxResults": 5,
    "prefix": "dapr-",
    "marker": "obstest",
    "delimiter": "jpg"
  }
}
```

The data parameters are:

- `maxResults` - (optional) sets the maximum number of keys returned in the response. By default the action returns up to 1,000 key names. The response might contain fewer keys but will never contain more.
- `prefix` - (optional) limits the response to keys that begin with the specified prefix.
- `marker` - (optional) marker is where you want Huawei OBS to start listing from. Huawei OBS starts listing after this specified key. Marker can be any key in the bucket. The marker value may then be used in a subsequent call to request the next set of list items.
- `delimiter` - (optional) A delimiter is a character you use to group keys. It returns objects/files with their object key other than that is specified by the delimiter pattern.

> 

Reviewer: Do you want to provide an example here of the `list` command?

Author: Sure, I will add an example for this.

Author: @msfussell Added example. pls review, thanks.
